### ISSUE TITLE
CatastrophicError: Don't know how to get type for type: @top

### ISSUE THREAD

**Issue by Marcono1234** [github/codeql#12664]
### Version
CodeQL extension version: 1.8.1 
CodeQL CLI version: 2.12.5 
Platform: win32 x64

### Description
I noticed by accident that in VS Code one of my query files had a red marker at the beginning of the file which said "An internal error occurred during compilation.". The "CodeQL Language Server" output of the VS Code extension contained the following:
```
{"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///#redacted#","diagnostics":[{"range":{"start":{"line":0,"character":0},"end":{"line":0,"character":0}},"severity":1,"source":"","message":"An internal error occurred during compilation."}]}}[2023-03-25 23:19:15] [ERROR] Unhandled error in QL frontend
com.semmle.util.exception.CatastrophicError: Don't know how to get type for type: @top
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getType(TypeCompiler.java:151)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getCharacterIfDefined(TypeCompiler.java:282)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getDomainType(TypeCompiler.java:241)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getType(TypeCompiler.java:141)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getCharacterType(TypeCompiler.java:329)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getType(TypeCompiler.java:143)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getClassType(TypeCompiler.java:293)
	at com.semmle.frontend.compiler.codegen.TypeCompiler.getType(TypeCompiler.java:139)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.getType(RelationCompiler.java:291)
	at com.semmle.frontend.compiler.codegen.FieldBuilder.buildField(FieldBuilder.java:54)
	at com.semmle.frontend.compiler.codegen.FieldBuilder.buildField(FieldBuilder.java:72)
	at com.semmle.frontend.compiler.codegen.relations.UserPredicateBuilder.getRelation(UserPredicateBuilder.java:99)
	at com.semmle.frontend.compiler.codegen.relations.UserPredicateBuilder.getRelation(UserPredicateBuilder.java:62)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.lambda$getRelation$3(RelationCompiler.java:194)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.lambda$computeRemappedAndCached$9(RelationCompiler.java:315)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.computeCached(RelationCompiler.java:347)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.computeRemappedAndCached(RelationCompiler.java:312)
	at com.semmle.frontend.compiler.codegen.relations.RelationCompiler.getRelation(RelationCompiler.java:188)
	at com.semmle.frontend.compiler.StageBuilder.lambda$makeStages$3(StageBuilder.java:119)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at com.semmle.frontend.compiler.StageBuilder.makeStages(StageBuilder.java:135)
	at com.semmle.frontend.compiler.StageBuilder.computeStages(StageBuilder.java:42)
	at com.semmle.frontend.compiler.QL2DILCompiler.createStages(QL2DILCompiler.java:184)
	at com.semmle.frontend.compiler.QL2DILCompiler.compile(QL2DILCompiler.java:86)
	at com.semmle.frontend.compiler.QLCompileTarget.createPipeline(QLCompileTarget.java:83)
	at com.semmle.frontend.server.AsynchronousDiagnosticReporter.doWork(AsynchronousDiagnosticReporter.java:144)
	at java.base/java.lang.Thread.run(Unknown Source)
```

However, the error is gone now and I haven't investigated if / how to reproduce it. Hopefully the stack trace is helpful for you nonetheless.

**Comment by alexet** [comment]
This should be fixed in the next release of CodeQL.

**Comment by alexet** [comment]
I think it will actually be fixed in `2.13.0` instead which will be the release after the next.
